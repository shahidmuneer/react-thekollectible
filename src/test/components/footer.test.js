import { render } from '../utils'
import Footer from '../../components/footer'
import React from 'react'
import Link from 'next/link'

describe('<Footer /> component', () => {
  it('renders with brand name and license description should not be blank', () => {
    const wrapper = render(<Footer />)
    expect(wrapper.getByText('conduit')).toBeDefined()
    expect(wrapper.container.getElementsByClassName('.attribution')).not.toBe('')
    expect(wrapper.asFragment()).toMatchSnapshot()
  })

  it('Render footer should anchor point to root ', () => {
    const { getByText } = render(<Footer />)

    expect(getByText('conduit').href).toBe('http://localhost/');
  })

  it('should contain a anchor with link to thinkster.io and Thinkster as link name in attribution span', () => {
    const wrapper = render(<Footer />)

    const thinksterAnchor = wrapper.container.querySelector('.attribution')

    expect(thinksterAnchor.querySelector('a').href).toBe('https://thinkster.io/')
    expect(thinksterAnchor.querySelector('a').text).toBe('Thinkster')
  })
})