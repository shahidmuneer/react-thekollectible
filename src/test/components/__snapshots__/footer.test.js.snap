// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> component renders with brand name and license description should not be blank 1`] = `
<DocumentFragment>
  <footer>
    <div
      class="container"
    >
      <a
        class="logo-font"
        href="/"
      >
        conduit
      </a>
      <span
        class="attribution"
      >
        An interactive learning project from 
        <a
          href="https://thinkster.io"
        >
          Thinkster
        </a>
        . Code & design licensed under MIT.
      </span>
    </div>
  </footer>
</DocumentFragment>
`;
